app/src/main/java/org/p2p/wallet/striga/ui/firststep/StrigaSignUpFirstStepFragment.kt

```diff
@@ -83,4 +107,10 @@ class StrigaSignUpFirstStepFragment :
             }
         }
     }
+
+    private fun onFragmentResult(requestKey: String, bundle: Bundle) {
+        if (requestKey != REQUEST_KEY) return
```

Comment:
this could be moved to `StrigaCountryPickerFragment` to hide actual `RESULT_KEY` (it should be private inside `StrigaCountryPickerFragment`)
StrigaSignUpFirstStepFragment should not manage result keys of non-owned code

```
private fun onFragmentResult(requestKey: String, bundle: Bundle) {
    if (requestKey != REQUEST_KEY) return
    val selectedCountry = StrigaCountryPickerFragment.getResult(bundle)
    presenter.onCountryChanged(selectedCountry ?: return)
}
```

```
class StrigaCountryPickerFragment {
  companion {
     fun getResult(bundle: Bundle): Country? {
         return bundle.getParcelableCompat(RESULT_KEY) as? Country
     }
  }
}
```

Reply:
I disagree with you, SignUpFirstFragment, has own REQUEST_KEY and RESULT_KEY, so it's just pass them to the picker fragment, and then just try to listen for result via own keys.
Why do you think, that's pickers result and request key ?